include/unifex/any_unique.hpp

```diff
@@ -272,15 +259,54 @@ struct _with_forwarding_tag_invoke<
   : _with_forwarding_tag_invoke<Derived, CPO, true, Ret(Args...)> {
 };
 
-inline const struct deallocate_cpo {
+struct _deallocate_cpo {
   using type_erased_signature_t = void(this_&&) noexcept;
 
-  template(typename T)
-      (requires tag_invocable<deallocate_cpo, T&&>)
+  template <typename T>
   void operator()(T&& obj) const noexcept {
-    tag_invoke(deallocate_cpo{}, (T &&) obj);
+    if constexpr (tag_invocable<_deallocate_cpo, T>) {
+      tag_invoke(_deallocate_cpo{}, (T &&) obj);
+    } else {
+      delete std::addressof(obj);
```

Comment:
When is this alternative code-path going to be hit?

The `_concrete_impl` class below always defines a _deallocate_cpo tag_invoke overload so will always call the allocator version?

Reply:
When you don't construct the `unique_any` with an allocator, I don't create a `_concrete_impl`. I `new` a `T` directly, which doesn't hook the `_deallocate_cpo`.

Reply:
Once this is merged I want to take a look at a godbolt of side-by-side assembly for any_unique constructed with no allocator and for one constructed with `std::allocator` (which should have the same semantics).
Would be enlightening to see where the extra code-size is going.